python/cairo-addons/src/cairo_addons/hints/maths.py

```diff
@@ -37,3 +37,19 @@ def is_positive_hint(ids: VmConsts):
 @register_hint
 def value_len_mod_two(ids: VmConsts):
     ids.remainder = ids.len % 2
+
+@register_hint
+def felt252_to_bits(ids: VmConsts, segments: MemorySegmentManager):
+    limb = ids.limb
+    current_len = ids.current_len
+    bits_ptr = ids.bits_ptr
+
+    # Convert the limb to binary representation and remove '0b' prefix
+    binary = bin(limb)[2:]
+
+    # Write each bit to memory
+    for i, bit in enumerate(binary):
+        segments.memory[bits_ptr + current_len + i] = int(bit)
+
+    # Update the current_len with the new length
+    ids.current_len = current_len + len(binary)
```

Comment:
- rename limb -> value (agnostic - it's just a felt value)
- current_len should not be required
- inlined intermediate bits_ptr variable
- used `load_data` to load a sequence of data at a given pointer
```suggestion
    # Convert the limb to binary representation and remove '0b' prefix
    # and write it to memory
    binary = bin(ids.value)[2:]
    segments.load_data(ids.bits_ptr, binary)
```

Reply:
note: update my suggestion sinppet to shift the value until it has the desired bit length (see similar byte function).